tests/test_proxy.py

```diff
@@ -5,49 +5,116 @@
 
 from urllib.parse import urlparse
 from jupyterhub.utils import exponential_backoff
-from tornado.httpclient import AsyncHTTPClient, HTTPRequest
+from tornado.httpclient import AsyncHTTPClient, HTTPRequest, HTTPClientError
+import json
+
 
 # Mark all tests in this file as asyncio
 pytestmark = pytest.mark.asyncio
 
 
-async def test_etcd_routing(etcd, clean_etcd, proxy, launch_backends):
-    routespec = ["/", "/user/first", "/user/second"]
-    target = ["http://127.0.0.1:9000", "http://127.0.0.1:9090", "http://127.0.0.1:9099"]
-    data = [{}, {}, {}]
-
-    routes_no = len(target)
+async def wait_for_services(proxy, target):
     traefik_port = urlparse(proxy.public_url).port
+    traefik_ip = urlparse(proxy.public_url).hostname
+    backend_port = urlparse(target).port
+    backend_ip = urlparse(target).hostname
 
-    # Check if traefik process is reacheable
+    # Wait until traefik and the backend are ready
     await exponential_backoff(
-        utils.check_host_up, "Traefik not reacheable", ip="localhost", port=traefik_port
+        utils.check_services_ready,
```

Comment:
Since we are parsing a url and then reconstructing it in check_services_ready, maybe check_services_ready should take a list of urls rather than separate ips and ports?